### DISCUSSION TITLE
ResizeObserver for Safari 12.x, 13.0.x

### DISCUSSION THREAD

**Discussion by stephent** [plouc/nivo#1908] [Q&A] [answered]
Since the ResizeObserver polyfill [was removed](https://github.com/plouc/nivo/commit/22f2bf70edb3c0b325e390fd9ff36ba2487b16bf#diff-ff689ec523614b53b1e19500f6ef6404afb91dc02870b3d52fa248b4daf2fed7), my project is broken for Safari 12 and 13 users when built with the current version of nivo:

```
    "@nivo/core": "^0.79.0",
    "@nivo/line": "^0.79.1",
```

I'm having trouble working out how to add back `resize-observer-polyfill` to my own project so that I can keep on the current version of nivo and get things working again.

I've tried the approach [here](https://github.com/JRJurman/rollup-plugin-polyfill/issues/7#issuecomment-637109188) without success - I still get this error:

```Object is not a constructor (evaluating 'new ResizeObserver((function(e){var t=e[0];return i(t.contentRect)}))')```

at line 11:43 of `../../src/hooks/useMeasure.js`

I guess I'm not including the polyfill correctly to have nivo be able to use it. 

Any ideas on how it should be done?

**Comment by plouc** [comment, marked as answer]
You could try this:

```
import ResizeObserver from 'resize-observer-polyfill'

window.ResizeObserver = ResizeObserver // set the observer globally before importing your app
```

**Reply by stephent** [reply]
Thank you - well, that was a simple fix!

**Comment by MarcCelani-at** [comment]
This is biting me as I try to upgrade as well. Should we open an issue for this?
